src/clzones.cpp

```diff
@@ -732,6 +832,188 @@ void zone_data::set_temporary_disabled( const bool enabled_arg )
+        return fd_magazines_loot_zone;
+    } else if( type == zone_type_LOOT_MANUALS ) {
+        return fd_manuals_loot_zone;
+    } else if( type == zone_type_LOOT_MAPS ) {
+        return fd_maps_loot_zone;
+    } else if( type == zone_type_LOOT_MA_MANUALS ) {
+        return fd_martial_arts_manuals_loot_zone;
+    } else if( type == zone_type_LOOT_MODS ) {
+        return fd_mods_loot_zone;
+    } else if( type == zone_type_LOOT_MUTAGENS ) {
+        return fd_mutagens_loot_zone;
+    } else if( type == zone_type_LOOT_OTHER ) {
+        return fd_other_loot_zone;
+    } else if( type == zone_type_LOOT_PDRINK ) {
+        return fd_perishable_drink_loot_zone;
+    } else if( type == zone_type_LOOT_PFOOD ) {
+        return fd_perishable_food_loot_zone;
+    } else if( type == zone_type_LOOT_SEEDS ) {
+        return fd_seeds_loot_zone;
+    } else if( type == zone_type_LOOT_SPARE_PARTS ) {
+        return fd_spare_parts_loot_zone;
+    } else if( type == zone_type_STRIP_CORPSES ) {
+        return fd_strip_corpses_loot_zone;
+    } else if( type == zone_type_LOOT_TOOL_MAGAZINE ) {
+        return fd_tool_magazines_loot_zone;
+    } else if( type == zone_type_LOOT_TOOLS ) {
+        return fd_tools_loot_zone;
+    } else if( type == zone_type_LOOT_TRAPS ) {
+        return fd_traps_loot_zone;
+    } else if( type == zone_type_UNLOAD_ALL ) {
+        return fd_unload_everything_loot_zone;
+    } else if( type == zone_type_LOOT_UNSORTED ) {
+        return fd_unsorted_loot_zone;
+    } else if( type == zone_type_LOOT_VEHICLE_PARTS ) {
+        return fd_vehicle_parts_loot_zone;
+    } else if( type == zone_type_LOOT_WEAPONS ) {
+        return fd_weapons_loot_zone;
+    } else if( type == zone_type_LOOT_WOOD ) {
+        return fd_wood_loot_zone;
+    } else if( type == zone_type_MINING ) {
+        return fd_mine_terrain_zone;
+    } else if( type == zone_type_MOPPING ) {
+        return fd_mop_tile_zone;
+    } else if( type == zone_type_NO_AUTO_PICKUP ) {
+        return fd_no_auto_pickup_zone;
+    } else if( type == zone_type_NO_NPC_PICKUP ) {
+        return fd_no_npc_pickup_zone;
+    } else if( type == zone_type_NPC_NO_INVESTIGATE ) {
+        return fd_npc_ignore_sounds_zone;
+    } else if( type == zone_type_NPC_INVESTIGATE_ONLY ) {
+        return fd_npc_investigation_area_zone;
+    } else if( type == zone_type_NPC_RETREAT ) {
+        return fd_npc_retreat_zone;
+    } else if( type == zone_type_SOURCE_FIREWOOD ) {
+        return fd_firewood_source_zone;
+    } else if( type == zone_type_ZONE_START_POINT ) {
+        return fd_start_point_zone;
+    } else if( type == zone_type_VEHICLE_DECONSTRUCT ) {
+        return fd_vehicle_deconstruct_zone;
+    } else if( type == zone_type_VEHICLE_PATROL ) {
+        return fd_vehicle_patrol_zone;
+    } else if( type == zone_type_VEHICLE_REPAIR ) {
+        return fd_vehicle_repair_zone;
+    } else {
+        return fd_null;
+    }
+}
+
+void zone_data::toggle_display()
```

Comment:
Current code does not seem to update the fields when the player changes the zone area.

To be honest I feel it might be better to draw the symbols and sprites within the rendering code directly, but it's up to you.

Reply:
Just realized this won't work if multiple zones with the same type overlap. Only fields outside remaining zones of the same type should be removed in that case.

Reply:
Overlapping zones of the same type will be an issue.